src/ring_theory/roots_of_unity.lean

```diff
@@ -119,29 +119,29 @@ end
 variables [comm_ring S]
 
 /-- Restrict a ring homomorphism between integral domains to the nth roots of unity -/
-def ring_hom.restrict_roots_of_unity (σ : R →+* S) (n : ℕ+) :
+def restrict_roots_of_unity [ring_hom_class F R S] (σ : F) (n : ℕ+) :
   roots_of_unity n R →* roots_of_unity n S :=
 let h : ∀ ξ : roots_of_unity n R, (σ ξ) ^ (n : ℕ) = 1 := λ ξ, by
 { change (σ (ξ : Rˣ)) ^ (n : ℕ) = 1,
-  rw [←σ.map_pow, ←units.coe_pow, show ((ξ : Rˣ) ^ (n : ℕ) = 1), from ξ.2,
-      units.coe_one, σ.map_one] } in
+  rw [←map_pow, ←units.coe_pow, show ((ξ : Rˣ) ^ (n : ℕ) = 1), from ξ.2,
+      units.coe_one, map_one σ] } in
 { to_fun := λ ξ, ⟨@unit_of_invertible _ _ _ (invertible_of_pow_eq_one _ _ (h ξ) n.2),
     by { ext, rw units.coe_pow, exact h ξ }⟩,
-  map_one' := by { ext, exact σ.map_one },
-  map_mul' := λ ξ₁ ξ₂, by { ext, rw [subgroup.coe_mul, units.coe_mul], exact σ.map_mul _ _ } }
+  map_one' := by { ext, exact map_one σ },
+  map_mul' := λ ξ₁ ξ₂, by { ext, rw [subgroup.coe_mul, units.coe_mul], exact map_mul σ _ _ } }
 
-@[simp] lemma ring_hom.restrict_roots_of_unity_coe_apply (σ : R →+* S) (ζ : roots_of_unity k R) :
-  ↑(σ.restrict_roots_of_unity k ζ) = σ ↑ζ :=
+@[simp] lemma restrict_roots_of_unity_coe_apply [ring_hom_class F R S] (σ : F)
+  (ζ : roots_of_unity k R) : ↑(restrict_roots_of_unity σ k ζ) = σ ↑ζ :=
 rfl
 
 /-- Restrict a ring isomorphism between integral domains to the nth roots of unity -/
 def ring_equiv.restrict_roots_of_unity (σ : R ≃+* S) (n : ℕ+) :
   roots_of_unity n R ≃* roots_of_unity n S :=
-{ to_fun := σ.to_ring_hom.restrict_roots_of_unity n,
-  inv_fun := σ.symm.to_ring_hom.restrict_roots_of_unity n,
+{ to_fun := restrict_roots_of_unity σ.to_ring_hom n,
+  inv_fun :=restrict_roots_of_unity σ.symm.to_ring_hom n,
```

Comment:
You no longer need these now, right?
```suggestion
{ to_fun := restrict_roots_of_unity σ n,
  inv_fun :=restrict_roots_of_unity σ.symm n,
```

Reply:
I would've thought so, but I don't think the instances are there yet; maybe this is because we don't have `equiv_like` yet. Lean says that it can't find a `ring_hom_class (R ≃+* S) ?m_1 ?m_2`.

Reply:
You won't need it once #11634 goes through, as it adds a coercion from `F` to `ring_hom` in the presence of `ring_hom_class`.